sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsyncQueryBenchmark.java

```diff
@@ -34,7 +71,7 @@ protected void onSuccess() {
 
     @Override
     protected void performWorkload(BaseSubscriber<FeedResponse<Document>> baseSubscriber, long i) throws InterruptedException {
-
+        concurrencyControlSemaphore.acquire();
```

Comment:
Why are we acquiring the lock so early in the `performWorkload`?

Reply:
sure, moved down. DONE